templates/vault/0/rancher-compose.yml

```diff
@@ -0,0 +1,67 @@
+version: '2'
+catalog:
+  name: Vault
+  version: 1.0-test-rancher1
+  description: |
+    Installs a Vault container, internal (Rancher) load-balancer, and allows connection to a Consul cluster
+  uuid: vault-rac-0
+  questions:
+    - variable: VAULT_LOCAL_CONFIG
+      label: Provide a default config?
+      description: |
+        This is the embedded JSON format config file that vault will use at startup
+      type: multiline
+      default: |
+        {"backend":{"consul":{"address":"http://<listenAddress>:<listenPort>", "advertise_addr":"http://<advertiseAddress>","path":"<Path>"}}, "listener":{"tcp":{"address":"0.0.0.0:18200","tls_disable":1}}}
+    - variable: useConsul
+      label: Would you like to back Vault with Consul?
+      description: |
+        Selecting yes or no allows you to back your vault repoository with a pre-installed
+        Consul cluster/server
+      default: false
+      type: boolean
+    - variable: consulService
+      label: Specify the Consul service to connect to
+      description: |
+        The pre-installed Consul server to back to
+      default: "select"
+      type: service
+    - variable: volumeDriver
+      label: Driver for volumes
+      description: |
+        How/Where to store your vault config and logs
+      default: local
+      type: enum
+      options:
+        - local
+        - rancher-nfs
+        - rancher-efs
+        - rancher-ebs
+services:
+  vault-lb:
+    scale: 1
+    start_on_create: true
+    lb_config:
+      certs: []
+      port_rules:
+      - priority: 1
+        protocol: tcp
+        service: vault
+        source_port: 8200
+        target_port: 18200
+      - priority: 2
+        protocol: tcp
+        service: vault
+        source_port: 8201
+        target_port: 8201

```

Comment:
Is mandatory expose vault cluster port?? If yes, What do you think if you add another question at catalog, to be able to choose `source_port` number to publish??

Have you consider to use `VAULT_CLUSTER_ADDR` as showed at  [vault image docker-entrypoint.sh](https://github.com/hashicorp/docker-vault/blob/75178223957b123d016181e628b0fe6443ab3d49/0.X/docker-entrypoint.sh)?? You could ask for a port to listen `VAULT_CLUSTER_PORT` and add variable into vault services docker definition
```
+      VAULT_CLUSTER_ADDR: http://0.0.0.0:${VAULT_CLUSTER_PORT}
```

and in this file, 
```
+        source_port: ${VAULT_CLUSTER_PORT}
+        target_port: ${VAULT_CLUSTER_PORT}
```

Reply:
Can't really accommodate this one due to the hard-coded port in the Vault container.  See my comments above for what should be a viable alternative.